### PR TITLE
Add JSON support

### PR DESCRIPTION
**Pull request by jenow**
## What does this PR do ?

- Add json support in websocket.
- Add unit tests

### How should this be manually tested?

```sh
./gradlew test
```

### CODE REVIEW

File: src/main/kotlin/io/kuzzle/sdk/coreClasses/responses/Response.kt [hunk 1]
```diff
@@ -92,7 +91,7 @@ class Response : Serializable {
     }
     requestId = kuzzleMap.getString("requestId")
     if (requestId == null) {
-      throw InternalException(KuzzleExceptionCode.MISSING_REQUESTID.message)
+      throw Exception(KuzzleExceptionCode.MISSING_REQUESTID.message)
```
Comment (Yoann-Abbes) [inline src/main/kotlin/io/kuzzle/sdk/coreClasses/responses/Response.kt]: :trollface:

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      install:
+        - cd $TRAVIS_BUILD_DIR/
+      script:
+        - ./gradlew test
+
+stages:
+  - name: Unit Tests
+    if: type =~ /(cron|push|pull_request)/ AND branch =~ /^master|[0-9]+-(dev|stable)$/
+  - name: Builds
+    if: type =~ /(cron|push|pull_request)/ AND branch =~ /^master|[0-9]+-(dev|stable)$/
```
Comment (alexandrebouthinon) [inline .travis.yml]: I think you do not need this stage for now :slightly_smiling_face:

File: src/main/kotlin/io/kuzzle/sdk/coreClasses/exceptions/KuzzleException.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package io.kuzzle.sdk.coreClasses.exceptions
+
+/**
+ * Root of all Kuzzle exceptions.
+ */
+open class KuzzleException : Exception {
+  /**
+   * @return The status code of the exception.
+   */
```
Comment (alexandrebouthinon) [inline src/main/kotlin/io/kuzzle/sdk/coreClasses/exceptions/KuzzleException.kt]: Is this an overlooked deletion? I don't know Kotlin's commenting standards

File: src/main/kotlin/io/kuzzle/sdk/coreClasses/exceptions/NotConnectedException.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package io.kuzzle.sdk.coreClasses.exceptions
+
+/**
+ * Thrown when attempting to interact with the network while not connected.
+ */
+open class NotConnectedException : KuzzleException {
+  /**
+   *
+   */
```
Comment (alexandrebouthinon) [inline src/main/kotlin/io/kuzzle/sdk/coreClasses/exceptions/NotConnectedException.kt]:
```suggestion
```

File: src/main/kotlin/io/kuzzle/sdk/coreClasses/json/ConcurrentHashMapTypeAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      }
+      `in`.endObject()
+      return map
+    }
+    return null
+  }
+
+  @Throws(IOException::class)
+  private fun writeObject(out: JsonWriter, value: Any?) {
+    if (value is Number) {
```
Comment (alexandrebouthinon) [inline src/main/kotlin/io/kuzzle/sdk/coreClasses/json/ConcurrentHashMapTypeAdapter.kt]: Kotlin has a pattern matching system that might work better for this use case (https://kotlinlang.org/docs/reference/control-flow.html#when-expression)
```kotlin
when (value) {
  is Number -> out.value(value as Number?)
  is Boolean -> out.value(value as Boolean)
  is ArrayList<*> -> { // you can declare scoped branch
    // The parsing logic you made...
  }
  else -> doSomethingElse() // You can also use 'else' 
}
```

Comment (jenow) [reply src/main/kotlin/io/kuzzle/sdk/coreClasses/json/ConcurrentHashMapTypeAdapter.kt]: I know about this pattern, I use it somewhere. I didn't write everything, I took the source from Java and let intellij transform everything, the goal being to make this SDK asap

Comment (alexandrebouthinon) [reply src/main/kotlin/io/kuzzle/sdk/coreClasses/json/ConcurrentHashMapTypeAdapter.kt]: Oh ok nevermind 🙂

File: .travis.yml (hunk 2 above)
Comment (jenow) [reply .travis.yml]: I leave it here so we don't forget it later

**Pull request merged**